src/database/database-leveldb.cpp

```diff
@@ -97,5 +98,100 @@ void Database_LevelDB::listAllLoadableBlocks(std::vector<v3s16> &dst)
 	delete it;
 }
 
-#endif // USE_LEVELDB
+AuthDatabaseLevelDB::AuthDatabaseLevelDB(const std::string &savedir)
+{
+	leveldb::Options options;
+	options.create_if_missing = true;
+	leveldb::Status status = leveldb::DB::Open(options,
+		savedir + DIR_DELIM + "auth.db", &m_database);
+	ENSURE_STATUS_OK(status);
+}
+
+AuthDatabaseLevelDB::~AuthDatabaseLevelDB()
+{
+	delete m_database;
+}
+
+bool AuthDatabaseLevelDB::getAuth(const std::string &name, AuthEntry &res)
+{
+	std::string raw;
+	leveldb::Status s = m_database->Get(leveldb::ReadOptions(), name, &raw);
+	if (!s.ok())
+		return false;
+	std::istringstream is(raw);
+
+	/*
+	u8 version = 0
+	std::string password
+	u16 number of privileges
+	for each privilege {
+		std::string privilege
+	}
+	s64 last_login
+	*/
+
+	if (readU8(is) != 0)
+		return false;
+
+	res.id = 1;
```

Comment:
is it safe to use a constant ID?
does the auth code (or anything else) use this?

Reply:
The plaintext database uses a constant ID.
https://github.com/minetest/minetest/blob/625b1005939e76d6a5b2d241635be3b8a89eb128/src/database/database-files.cpp#L263-L269